### PR TITLE
added algo

### PR DESCRIPTION
**Pull request by KunalShivnekar**

### CODE REVIEW

File: src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ import arraysandlists.mergeTwoLists
 import arraysandlists.reverseList
```
Comment (KunalShivnekar) [inline src/main/kotlin/Main.kt:3]: It's generally recommended to move test calls like this to a dedicated test file (e.g., `src/test/kotlin/leetcode/CheckValidCutsTest.kt`) to keep the `main` function clean and focused on application entry point.

File: src/main/kotlin/leetcode/othersolutions.kt [hunk 2]
```diff
@@ -83,7 +83,7 @@ fun getProfit(prices: IntArray, index:Int, mem:IntArray):Int{
     for(j in index+1 until prices.size){
```
Comment (KunalShivnekar) [inline src/main/kotlin/leetcode/othersolutions.kt:83]: This function checks if two valid cuts can be made to divide the given area, considering the provided rectangles. It iterates through possible horizontal and vertical cut positions.

File: src/main/kotlin/leetcode/othersolutions.kt [hunk 3]
```diff
@@ -83,7 +83,7 @@ fun getProfit(prices: IntArray, index:Int, mem:IntArray):Int{
...
         profit = Math.max(profit, prices[j]-prices[index] + getProfit(prices, j+2, mem))
     }
```
Comment (KunalShivnekar) [inline src/main/kotlin/leetcode/othersolutions.kt:85]: Counts rectangles that are entirely to the left (or below for vertical cuts) of the current cut position.

File: src/main/kotlin/leetcode/othersolutions.kt [hunk 4]
```diff
@@ -83,7 +83,7 @@ fun getProfit(prices: IntArray, index:Int, mem:IntArray):Int{
...
-    profit = Math.max(profit,getProfit(prices, index+1, mem))
```
Comment (KunalShivnekar) [inline src/main/kotlin/leetcode/othersolutions.kt:86]: Counts rectangles that are entirely to the right (or above for vertical cuts) of the current cut position.

File: src/main/kotlin/leetcode/othersolutions.kt [hunk 5]
```diff
@@ -83,7 +83,7 @@ fun getProfit(prices: IntArray, index:Int, mem:IntArray):Int{
...
+    profit = profit.coerceAtLeast(getProfit(prices, index + 1, mem))
```
Comment (KunalShivnekar) [inline src/main/kotlin/leetcode/othersolutions.kt:86]: Indicates if any rectangle spans across the current cut position, making it an invalid cut.

File: src/main/kotlin/leetcode/othersolutions.kt [hunk 6]
```diff
@@ -109,4 +109,58 @@ fun commonChars(words: Array<String>): List<String> {
...
+        cuts++
+        for(rect in rectangles) {
+            if(rect[1]<i && rect[3]>i){
+                cuts--
+                flag = true
+                break
+            }else {
+                if(rect[1]>=i)
+                    r++
+                else
```
Comment (KunalShivnekar) [inline src/main/kotlin/leetcode/othersolutions.kt:155]: Consider adding more comprehensive unit tests for this function to cover various edge cases and scenarios, such as empty rectangles array, `n` values, and different rectangle configurations.

**Review by KunalShivnekar — COMMENTED** [review]
Here are some suggestions for your pull request:
